client/extensions/hello-dolly/state/test/reducer.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import reducer from '../reducer';
+import {
+	HELLO_DOLLY_NEXT_LYRIC,
+} from '../action-types';
+
+describe( 'reducer', () => {
+	it( 'should have initial state', () => {
+		const state = reducer( undefined, { type: undefined } );
+
+		expect( state.lyricIndex ).to.be.a( 'number' );
+	} );
```

Comment:
I'll just poke here and point out that because of how we used an object as our state instead of a primitive data type it made us feel like we needed to test this. however, if we stick to the simpler reducers and then combine them with `combineReducers()` (when needing more than one property at a given subtree) then these tests have already been rigorously made and vetted in Redux.

in this sense we end up less with a unit test here and more with an integration test or feel-good test telling us we didn't botch something up

notice also that because we have jumped into the fringe of what unit testing means for us we have designed something to test the inner workings of how our function works and have overlooked the _behavior_ which the unit test is designed to assert - we've dug _into_ the black box instead of guarantee its interface

```js
it( 'should initialize to the first index', () => {
	expect( reducer( undefined, { type: '@@test/INIT' } ).lyricIndex ).to.equal( 0 );
} );
```

I also included a string value for the action type because I believe that there can be problems with some reducers when given `undefined`